File: AssetManagement.Application/Assets/UpdateAsset/UpdateAssetRequest.cs
using AssetManagement.Domain.Entities.Assets;


namespace AssetManagement.Application.Assets.UpdateAsset
{
    public sealed record UpdateAssetRequest(
        string AssetName,
        AssetType AssetType,
        AssetStatus AssetStatus,
        string Manufacturer,
        string Model,
        string? MacAddress,
        string? WifiMacAddress,
        string? Imei,
        string? OperatingSystem,
        string? OperatingSystemVersion,
        Guid AssignedStoreId,
        Guid? AssignedUserId,
        Guid? AssignedVendorId,
        string? RfIdTagId
        );
}

File: Client/src/app/features/assets/data-access/models/update-asset-request.ts
import { AssetStatus } from "./asset-status";
import { AssetType } from "./asset-type";

export interface UpdateAssetRequest {
  readonly assetName: string | null;
  readonly assetStatus: AssetStatus;
  readonly assetType: AssetType;
  readonly manufacturer: string | null;
  readonly model: string | null;
  readonly serialNumber: string;
  readonly macAddress: string | null;
  readonly wifiMacAddress: string | null;
  readonly imei: string | null;
  readonly assignedStoreId: string | null;
  readonly operatingSystem: string | null;
  readonly operatingSystemVersion: string | null;
  readonly assignedVendorId: string | null;
  readonly rfidTagId: string | null;
}
